core/src/main/java/io/questdb/cairo/ColumnPurgeOperator.java

```diff
@@ -86,10 +86,9 @@ public void close() throws IOException {
             longBytes = 0;
         }
         closePurgeLogCompleteFile();
-        path.close();
-        if (txnScoreboard != null) {
-            txnScoreboard.close();
-        }
+        Misc.free(path);
+        txReader = Misc.free(txReader);
```

Comment:
Why you decided to close `txReader` here? This class does not always own `txReader`, sometimes it borrows it for processing

Reply:
yes, thank you.